### PR TITLE
feat: ability to customize SelectionDot component

### PR DESCRIPTION
**Pull request by robinvw1**
# Description

The current selection dot is 'design-wise' quite opinionated. It has a light shadow and an outer ring.

Instead of introducing a lot of new props (no offence https://github.com/margelo/react-native-graph/pull/21) to customize the look-and-feel of the selection dot, I went for a flexible way to provide your own SelectionDot component.

# Demo

Default situation | With custom selection dot
--- | ---
<video src="https://user-images.githubusercontent.com/793157/181266279-36320c71-f76e-4a8b-9794-fa9f0be24f5b.mp4"> | <video src="https://user-images.githubusercontent.com/793157/181266914-707907b8-39af-4a3b-b286-af1c354b60de.mp4">

### CODE REVIEW

File: src/SelectionDot.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  )
+
+  const setIsActive = useCallback(
+    (active: boolean) => {
+      runSpring(circleRadius, active ? 5 : 0, {
+        mass: 1,
+        stiffness: 1000,
+        damping: 50,
+        velocity: 0,
+      })
```
Comment (mrousavy) [inline src/SelectionDot.tsx:32]: I'd move that into the useAnimatedReaction, no need to add an extra callback here

File: src/AnimatedLineGraph.tsx [hunk 2]
```diff
@@ -227,6 +217,13 @@ export function AnimatedLineGraph({
     ],
     [pathEnd]
   )
+  const dotProps: SelectionDotProps = {
+    isActive,
+    color,
+    lineThickness,
+    circleX,
+    circleY,
+  }
```
Comment (mrousavy) [inline src/AnimatedLineGraph.tsx:226]: does it make sense to create them here and spread them afterwards again? Why not just pass directly to the View?

**Review by mrousavy — APPROVED** [review]
Looks pretty good in general! Just left two comments (questions)

File: src/AnimatedLineGraph.tsx (hunk 2 above)
Comment (robinvw1) [reply src/AnimatedLineGraph.tsx]: fixed in https://github.com/margelo/react-native-graph/pull/35/commits/b61540f3b0cbc26594c094feba930dac090e3b92

**Comment by mrousavy** [comment]
LGTM thank you!

**Pull request merged**